source/common/http/header_map_impl.cc

```diff
@@ -346,6 +360,11 @@ void HeaderMapImpl::addReferenceKey(const LowerCaseString& key, const std::strin
 }
 
 void HeaderMapImpl::addCopy(const LowerCaseString& key, uint64_t value) {
+  auto* entry = getExistingInline(key.get().c_str());
+  if (entry) {
+    appendToHeader(entry->value(), fmt::format("{}", value));
```

Comment:
perf nit: fmt::format is pretty slow. I think we could make this substantially faster by doing itoa directly inline during the append (special case of append into a header string). At least maybe put a perf TODO here?